src/clr-angular/forms/common/if-error/if-error.service.ts

```diff
@@ -27,8 +27,10 @@ export class IfErrorService implements OnDestroy {
     // Wait for the control to be available
     this.subscriptions.push(
       this.ngControlService.controlChanges.subscribe(control => {
-        this.control = control;
-        this.listenForChanges();
+        if (control) {
```

Comment:
Is this guard redundant? What I mean by that is I haven't heard of a subscribe callback being called without an argument. Is that possible?

Reply:
In the control components, the NgControl is injected with `@Optional()` and if there was no control it will set a `null` value through the observable. So yes this is needed, but should probably guard as well at the observable level. Since injection is runtime we can't be certain that we'll get the value or not.